File: VB/ChildrenPath/ViewModel.vb
Imports System.Collections.Generic

Namespace ChildrenPath

    Friend Class ViewModel

        Public Property AppMenu As Menu

        Public Property SelectedItem As MenuItem

        Public Sub New()
            Me.AppMenu = New ChildrenPath.Menu()
            Me.SelectedItem = Me.AppMenu.MenuItems(0)
        End Sub
    End Class

    Public Class Menu

        Public Property MenuItems As List(Of ChildrenPath.MenuItem)

        Public Sub New()
            Me.MenuItems = ChildrenPath.Menu.GetMenuItems()
        End Sub

        Private Shared Function GetMenuItems() As List(Of ChildrenPath.MenuItem)
            Dim items As System.Collections.Generic.List(Of ChildrenPath.MenuItem) = New System.Collections.Generic.List(Of ChildrenPath.MenuItem)()
            Dim subitems As System.Collections.Generic.List(Of ChildrenPath.MenuItem) = New System.Collections.Generic.List(Of ChildrenPath.MenuItem)()
            subitems.Add(New ChildrenPath.MenuItem() With {.Caption = "SubItem3"})
            items.Add(New ChildrenPath.MenuItem() With {.Caption = "Item1", .SubItems = New System.Collections.Generic.List(Of ChildrenPath.MenuItem)() From {New ChildrenPath.MenuItem() With {.Caption = "SubItem1"}, New ChildrenPath.MenuItem() With {.Caption = "SubItem2", .SubItems = subitems}}})
            items.Add(New ChildrenPath.MenuItem() With {.Caption = "Item2", .SubItems = New System.Collections.Generic.List(Of ChildrenPath.MenuItem)() From {New ChildrenPath.MenuItem() With {.Caption = "SubItem1"}, New ChildrenPath.MenuItem() With {.Caption = "SubItem2"}}})
            Return items
        End Function
    End Class

    Public Class MenuItem

        Public Property Caption As String

        Public Property SubItems As List(Of ChildrenPath.MenuItem)
    End Class
End Namespace

File: CS/ChildrenPath/ViewModel.cs
using System.Collections.Generic;

namespace ChildrenPath {

    class ViewModel {
        public Menu AppMenu { get; set; }
        public MenuItem SelectedItem { get; set; }
        public ViewModel() {
            AppMenu = new Menu();
            SelectedItem = AppMenu.MenuItems[0];
        }
    }

    public class Menu {
        public List<MenuItem> MenuItems { get; set; }
        public Menu() {
            MenuItems = GetMenuItems();
        }
        private static List<MenuItem> GetMenuItems() {
            List<MenuItem> items = new List<MenuItem>();
            List<MenuItem> subitems = new List<MenuItem>();
            subitems.Add(new MenuItem() { Caption = "SubItem3" });
            items.Add(new MenuItem() {
                Caption = "Item1",
                SubItems = new List<MenuItem>() { new MenuItem() { Caption = "SubItem1" },
                    new MenuItem() { Caption = "SubItem2", SubItems=subitems }
                }
            });
            items.Add(new MenuItem() {
                Caption = "Item2",
                SubItems = new List<MenuItem>() { new MenuItem() { Caption = "SubItem1" },
                    new MenuItem() { Caption = "SubItem2" }
                }
            });
            return items;
        }
    }

    public class MenuItem {
        public string Caption { get; set; }
        public List<MenuItem> SubItems { get; set; }
    }
}
